engine/order_manager.go

```diff
@@ -284,6 +284,49 @@ func (m *OrderManager) validate(newOrder *order.Submit) error {
 	return nil
 }
 
+// Modify depends on the order.Modify.ID and order.Modify.Exchange fields to uniquely
+// identify an order to modify.
+func (m *OrderManager) Modify(mod *order.Modify) (*order.ModifyResponse, error) {
+	if m == nil {
+		return nil, fmt.Errorf("order manager %w", ErrNilSubsystem)
+	}
+	if atomic.LoadInt32(&m.started) == 0 {
+		return nil, fmt.Errorf("order manager %w", ErrSubSystemNotStarted)
+	}
+
+	exch := m.orderStore.exchangeManager.GetExchangeByName(mod.Exchange)
+	if exch == nil {
+		return nil, ErrExchangeNotFound
+	}
+
+	id, err := exch.ModifyOrder(mod)
+	if err != nil {
+		return nil, err
+	}
+
+	// We now have to reflect this change in GCT's internal state of managed orders.
+
+	// TODO: Current implementation is here for illustrative purposes.  It introduces
+	// race conditions.
+	det, err := m.orderStore.getByExchangeAndID(mod.Exchange, mod.ID)
+	if err != nil {
+		// TODO: How to handle that?  The order is updated on the
```

Comment:
Would need to fetch or build an order.Detail from order.Modify.

Reply:
Could you please elaborate a bit further on that? The `Modify struct` has just partial information at this point, I don't see how we can build a `Detail` reliably out of that.. Plus there *has to be* a managed Detail already, right?. I was thinking of panic()ing here as it's a case we should never hit. How bad would that be? :)

Reply:
You are correct, it can't reliable generate an order detail. 

> Plus, there has to be a managed Detail already, right?

There definitely should be but if we don't sync orders fast enough this might be a very unlikely scenario. I would even opt to 1) Add in a error log output and then retrieve the order from the exchange (albeit this is slow and adds http.Request friction) to then build out the order detail. The probability of this actually hitting and executing is low. 

> I was thinking of panic()ing here as it's a case we should never hit. How bad would that be? 

We try not to introduce panics in this codebase, I do see why you would put one there though. Its been a convention for a while to keep the program running and logging.

Reply:
In such a scenario, it may be best to still create the order in the order manager when it receives an order.Modify for an order it does not yet have knowledge about. - it having knowledge that an order exists is still very important information, even if its just an orderID and its asset type. Then as Shazbert said, fetch order details afterwards separately in a go routine to update the sparse order